docs/05-infrastructure-nodes.md

```diff
@@ -0,0 +1,333 @@
+# OpenShift Infrastructure Nodes
+The OpenShift subscription model allows customers to run various core
+infrastructure components at no additional charge. In other words, a node
+that is only running core OpenShift infrastructure components is not counted
+in terms of the total number of subscriptions required to cover the
+environment.
+
+OpenShift components that fall into the infrastructure categorization
+include:
+
+* kubernetes and OpenShift control plane services ("masters")
+* router
+* container image registry
+* cluster metrics collection
+* cluster aggregated logging
+* service brokers
+
+Any node running a container/pod/component not described above is considered
+a worker and must be covered by a subscription.
+
+## More MachineSet Details
+In [the cluster-scaling excercises](04-scaling-cluster.md) you explored the
+use of `MachineSets` and adding replicas within them. In the case of an
+infrastructure node, we want to create additional `Machines` that have
+specific kubernetes labels. Then, we can configure the various components to
+run specifically on nodes with those labels.
+
+To accomplish this, you will create additional `MachineSets`. The easiest way
+to do this is to `get` the existing `MachineSets` into a file, and then
+modify them. This is because the `MachineSet` has some details that are
+specific to the AWS region that the cluster is deployed in, like the AWS EC2
+AMI ID. For example, given the following output of `oc get machineset -n
+openshift-cluster-api`:
+
+    NAME                         DESIRED   CURRENT   READY     AVAILABLE   AGE
+    190125-3-worker-us-west-1b   2         2         2         2           3h
+    190125-3-worker-us-west-1c   1         1         1         1           3 
+
+There are two available EC2 AZs into which we can deposit infrastructure
+components (`1b` and `1c`). Take a look at one specifically with `oc get
+machineset 190125-3-worker-us-west-1b -n openshift-cluster-api -o yaml`.
+There are a few very important sections.
+
+### Metadata
+The `metadata` on the `MachineSet` itself includes information like the name
+of the `MachineSet` and various labels:
+
+```YAML
+metadata:
+  creationTimestamp: 2019-01-25T16:00:34Z
+  generation: 1
+  labels:
+    sigs.k8s.io/cluster-api-cluster: 190125-3
+    sigs.k8s.io/cluster-api-machine-role: worker
```

Comment:
The cluster-api stack reads only `sigs.k8s.io/cluster-api-machine-type` label to determine if a machine manages a master node or not. On the other hand, `sigs.k8s.io/cluster-api-machine-role` is ignored. So unless it is used by another component, it has no value and can be removed. @enxebre WDYT? The role label was used in the past to get the right security groups/subnets. Though, we have refactored that code out in favor of `awsproviderconfig` in the `providerSpec` field.